kernel/src/process.rs

```diff
@@ -1649,103 +1594,10 @@ struct LoadResult {
-
-            // Total size of fixed segment
-            let aligned_fixed_len = align8!(aligned_stack_len + pic_values.data_size +
-                                            pic_values.got_size + pic_values.bss_size);
-
-            // Verify target data fits in memory before writing anything
-            if (aligned_fixed_len) > mem_size as u32 {
-                // When a kernel warning mechanism exists, this panic should be
-                // replaced with that, but for now it seems more useful to bail out to
-                // alert developers of why the app failed to load
-                panic!("{:?} failed to load. Stack + Data + GOT + BSS ({}) > available memory ({})",
-                       tbf_header.get_package_name(flash_start_addr),
-                       aligned_fixed_len,
-                       mem_size);
-            }
-
-            // Copy the GOT and data into base memory
-            for (orig, dest) in got.iter().chain(data.iter()).zip(got_andthen_data_ram.iter_mut()) {
-                *dest = *orig
-            }
-
-            // Zero out BSS
-            intrinsics::write_bytes(bss, 0, pic_values.bss_size as usize);
-
-            // Helper function that fixes up GOT entries
-            let fixup = |addr: &mut u32| {
-                let entry = *addr;
-                if (entry & 0x80000000) == 0 {
-                    // Regular data (memory relative)
-                    *addr = entry + (got_base as u32);
-                } else {
-                    // rodata or function pointer (code relative)
-                    *addr = (entry ^ 0x80000000) + (text_start as u32);
-                }
-            };
-
-            // Fixup Global Offset Table
-            let mem_got: &mut [u32] = slice::from_raw_parts_mut(got_base as *mut u32,
-                                                                (pic_values.got_size as usize) /
-                                                                mem::size_of::<u32>());
-
-            for got_cur in mem_got {
-                fixup(got_cur);
-            }
-
-            // Fixup relocation data
-            for (i, addr) in rel_data.iter().enumerate() {
-                if i % 2 == 0 {
-                    // Only the first of every 2 entries is an address
-                    fixup(&mut *(got_base.offset(*addr as isize) as *mut u32));
-                }
-            }
-
-            let load_result = LoadResult {
-                // Since we set these up on behalf of the process we know right
-                // where the are.
-                initial_stack_pointer: mem_base.offset(aligned_stack_len as isize),
-                initial_sbrk_pointer: mem_base.offset(aligned_fixed_len as isize),
-                header: tbf_header,
-            };
-
-            Some(load_result)
-        } else {
-            // If we don't have any pic values, we can't do the fixup.
-            None
-        }
+        unimplemented!("Sorry, no more pic fixup");
```

Comment:
Maybe put something slightly more helpful here. Perhaps a pointer to an as-yet unwritten blog post?

> "Kernel PIC fixup has been deprecated and removed. See https://www.tockos.org/blog/2018/userspace-pic" for more information"

That'll let us give a longer explanation of what changed, as well as remediation steps (update userland and rebuild probably sufficient)